gradle/scripts/lib/java-javadoc.gradle

```
@@ -316,126 +341,160 @@ allprojects {
-                                        def packageNamePattern = /^\p{javaJavaIdentifierStart}\p{javaJavaIdentifierPart}*(?:\.\p{javaJavaIdentifierStart}\p{javaJavaIdentifierPart}*)*$/
-                                        return line.matches(packageNamePattern)
-                                    }
-                                }
-                                success = tmpPackageListFile.renameTo(packageListFile)
+                        // Java older than 10 does not support element-list.
+                        // No problem. We can generate package-list from element-list.
+                        def tmpPackageListFile = new File("${packageListFile}.tmp")
+                        tmpPackageListFile.withWriter('utf-8') { out ->
+                            // Find only the lines with a package name.
+                            elementListFile.filterLine(out, 'utf-8') { line ->
+                                def packageNamePattern = /^\p{javaJavaIdentifierStart}\p{javaJavaIdentifierPart}*(?:\.\p{javaJavaIdentifierStart}\p{javaJavaIdentifierPart}*)*$/
+                                return line.matches(packageNamePattern)
                             }
                         }
+                        success = tmpPackageListFile.renameTo(packageListFile)
                     }
                 }
-
-                if (success) {
-                    linksOffline javadocUrl, "${listFileDir}"
-                }
             }
+        }
 
-            if (JavaVersion.current() >= JavaVersion.VERSION_11) {
-                // Javadoc in Java 11+ has more strict checks related with module system,
-                // so we have to use Java 11+ API docs.
-                addOfflineLink('java11', 'https://docs.oracle.com/en/java/javase/11/docs/api/')
+        return [success, javadocUrl, listFileDir]
+    }
+
+    def downloadListFile(File listFile, URL listUrl) {
+        // Do not attempt to download more than once.
+        if (!visitedUrls.add(listUrl.toString())) {
+            return
+        }
+
+        def success = false
+        def tmpListFile = new File("${listFile}.tmp")
+
+        listFile.parentFile.mkdirs()
+        listFile.delete()
+
+        logger.lifecycle("Download ${listUrl}")
+
+        try {
+            // Set some fake headers for the web sites who blocks a URLConnection.
+            def conn = listUrl.openConnection() as HttpURLConnection
+            conn.setRequestProperty("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")
+            conn.setRequestProperty('Accept-Encoding', 'identity')
+            conn.setRequestProperty("Accept-Language", 'en-US,en;q=0.5')
+            conn.setRequestProperty("Cache-Control", 'no-cache')
+            conn.setRequestProperty('Pragma', 'no-cache')
+            conn.setRequestProperty('User-Agent', "Gradle/${gradle.gradleVersion} (${project.group}:${project.ext.artifactId})")
```

`${gradle.gradleVersion}` was invisible from this function when I locally run `./gradlew --rerun-tasks --no-build-cache clean offlineDownloadTask`
```
groovy.lang.MissingPropertyException: Could not get unknown property 'gradle' for task ':offlineDownloadTask' of type OfflineLinkDownloadTask.
```
; and it was weird that I couldn't rerun the failed task without explicitly disabling cache.
```suggestion
            conn.setRequestProperty('User-Agent', "Gradle/${project.gradle.gradleVersion} (${project.group}:${project.ext.artifactId})")
```

---

This should be fixed 😅   Thanks for pointing this out